### ISSUE TITLE
consider renaming the library

### ISSUE THREAD

**Issue by glyph** [alecthomas/voluptuous#287]
I didn't see anyone having raised this issue before, so I thought perhaps you might not realize: the word "voluptuous" in english has a heavily sexual connotation.  [The very first noun associated with it in the dictionary definition](https://www.merriam-webster.com/dictionary/voluptuous) is "nudes".  

[Using unnecessarily sexualized language can create an environment that is unduly hostile to women](http://geekfeminism.wikia.com/wiki/Sexualized_environment), even if this is not intended. Many of the biases against women that such language provokes are totally unconscious, so even if the intent is not harmful, it can be bad regardless.

Finally, the _very first_ thing in the README says "despite the name" so it seems that perhaps it's occurred to you that this isn't a great name for a library.

In summary: in the interests of creating a more inclusive community, please come up with a new name for this library.  Thank you for reading!

**Comment by EgoWumpus** [comment]
I want to strongly second this. Dictionary definitions aside, many of us use python in a professional environment, that frowns poorly on anything remotely sexually connoted, nevermind strongly so. An informal poll of my female colleagues was overwhelmingly unanimous. I'm going to mince words a bit here, but it was wildly unpopular. One straight up said she'd never use the library.

And, personally, I think that's a shame because this is a very strong, very useful library, solving a real problem in an elegant way. Moreover the community here has been a respectful one, inclusive of women and I haven't seen any instances of problematic behavior. Literally the only problem is the name: and it seems eminently fixable.

Naming is the hardest thing, so I'll go so far as to suggest `morphology`, which attends to nature of the package and doesn't currently seem to be in use, to get the brainstorm started.

If this is an issue of time I'm willing to commit to doing the code update and shepherding the package name update. 

Thanks to Alec and all the contributors for taking the time to consider this, and all the work they've done on the package.

**Comment by glyph** [comment]
To echo @EgoWumpus's concern here as well: I filed this issue on behalf of actual women finding the name upsetting as well; I phrased it as if it were a hypothetical or abstract concern, but there are definitely real people who find the name offensive and don't use the library as a result.

**Comment by alecthomas** [comment]
While I understand where you're coming from, I will not change the name of this library. *Most* words are not offensive in and of themselves, but become so from context. Some are (racist words, specific sexual words, etc.), but this is not one of them. It is a beautiful word in all its meanings (the [second](http://dictionary.cambridge.org/dictionary/english/voluptuous) is why I chose it), but it seems in the minds of some its usage is now solely the domain of smut. I reject that appropriation. Perhaps that is anachronistic.

It's also genuinely surprising to me that a word like *this*, devoid of any sexual context, could be found offensive. Perhaps the secondary meanings of the word is not in common usage in the US? Perhaps the only interactions those who are offended have had with the word have been through smutty media. If that is the case, that is unfortunate.

As an aside, thank you for being civil, I appreciate it. The only time anybody else has brought this up **ever** before, it was a member of the PSF who was highly aggressive and offensive, to the point where he's the only person I have ever blocked on Twitter. He was going through some pretty heavy personal problems, but still.

**Comment by aurynn** [comment]
Hi;

I'm the woman the @glyph referred to.

Unfortunately, the sexualised meaning of `voluptuous` isn't anachronistic; this is the common meaning. This is problematic and offensive because it acts to sexualise otherwise professional spaces, and directly alludes to women's bodies, not a generic meaning of beauty.

Comparably problematic & offensive library names would be examples from the Ruby community, such as `girlfriend` or `trollop`.

Because of the sexualised connotation and the inappropriateness in professional settings, this puts me in the position of having to fork downstream projects to use `schema` instead.

**Comment by glyph** [comment]
> While I understand where you're coming from, I will not change the name of this library.

That's very disappointing.

> Most words are not offensive in and of themselves, but become so from context. Some are (racist words, specific sexual words, etc.), but this is not one of them. It is a beautiful word in all its meanings (the second is why I chose it), but it seems in the minds of some its usage is now solely the domain of smut. I reject that appropriation. Perhaps that is anachronistic.

What you're doing here is assuming that the conduit metaphor (https://en.wikipedia.org/wiki/Conduit_metaphor) is a literally true representation of language; that you, putting your own idea of the word into your communications, have voided its problematic meaning, because those who hear the word will "unpack" your intention.

This is not, as I suspect you believe, an idiosyncratic affectation, it's an error in thinking (one that has been extensively taxonomized by linguists and neuroscientists).

Let me put that in a less fancy way.  Look at the images you see on this page: https://www.google.com/search?q=voluptuous&tbm=isch and tell me that it is just "in the minds of some" that this word references smut.

> It's also genuinely surprising to me that a word like this, devoid of any sexual context, could be found offensive. Perhaps the secondary meanings of the word is not in common usage in the US?

It's not *common* in the UK either, or for that matter (from what I can tell) Australia, India, New Zealand, or South Africa; I took care to do some research first into non-US usages.  You've somehow insulated yourself from the popular meaning of the word in all english-speaking cultures.

For one example, here's an article in the Telegraph, which is most definitely not a US publication, listing this word in a list of words "only used to describe women": http://www.telegraph.co.uk/women/life/ambitious-frigid-and-frumpy-25-words-we-only-use-to-describe-wom/

> As an aside, thank you for being civil, I appreciate it.

I thought it was best to give you the benefit of the doubt, assuming that you were unaware of the impact that your poor choice of naming was having.  While @aurynn was the initial complainant, I did run this by a few other women to just check, and it would be fair to characterize the reaction as "unanimous" as @EgoWumpus did.  At least one other woman in question immediately asked what alternative libraries existed.

As I said, now that you *do* know that you've chosen an upsetting and professionally inappropriate word, I find it disappointing that you didn't even *consider* changing the name.
On behalf of other women who might be confronted with usages of this library, I earnestly hope you will reconsider.

**Comment by alecthomas** [comment]
I definitely don't want anyone to be offended, so please do find another library to use. Here are some options: [good](https://pypi.python.org/pypi/good/0.0.7-0) has a similar API and [schema](https://pypi.python.org/pypi/schema/0.6.6) has a similar approach, and seems quite popular. All the best.
